[webkitpy] Increase the default number of child processes spawned for arm64 Macs

ryanhaddad opened the pull request:
#### b0febf1cc635f1e2118218ff99b19c2fa31767c3
<pre>
[webkitpy] Increase the default number of child processes spawned for arm64 Macs
<a href="https://bugs.webkit.org/show_bug.cgi?id=248625">https://bugs.webkit.org/show_bug.cgi?id=248625</a>
rdar://102873021

Reviewed by Jonathan Bedard and Geoffrey Garen.

arm64 Mac hardware can handle more WebKitTestRunner processes than the
default number we calculate. Local testing suggests that doubling the count
yields an 18 minute speedup in overall test runtime on M1 hardware, so lets
hardcode this increase until we implement a more sophisticated test scheduling
mechanism.

* Tools/Scripts/webkitpy/port/mac.py:
(MacPort.default_child_processes):

Canonical link: <a href="https://commits.webkit.org/257754@main">https://commits.webkit.org/257754@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/bac5d7562e2efd0e21a0213aefa2feffac6a450c

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/98847 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/8058 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/32119 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/108262 "Built successfully") | [✅ 🛠 🧪 win](https://ews-build.webkit.org/#/builders/10/builds/168519 "Built successfully and passed tests") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/102801 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/8609 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/85425 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/91380 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/106250 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/104530 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/6519 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/90187 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/33548 "Passed tests") | 
| [❌ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/5/builds/102374 "Found 1 webkitpy python2 test failure: webkitpy.port.server_process_unittest.TestServerProcess.test_broken_pipe") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/88341 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/21532 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/76413 "Found 1 new API test failure: /WebKitGTK/TestWebKitAccessibility:/webkit/WebKitAccessibility/accessible/state (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/1969 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/23048 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/1878 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/45420 "Found 1 new test failure: media/video-inactive-playback.html (failure)") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/5322 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/6834 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/42522 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/3277 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

geoffreygaren commented on Tools/Scripts/webkitpy/port/mac.py:
```diff
@@ -225,6 +225,11 @@ def operating_system(self):
     def default_child_processes(self, **kwargs):
         default_count = super(MacPort, self).default_child_processes()
 
+        # FIXME: arm64 Mac hardware can handle more processes than the default number we calculate.
```
Intel Macs were already implicitly doubling due to hyper threading (counting 2 logical cores for each physical core). That might explain why explicitly doubling for arm64 Macs (which don't have hyper-threading) roughly works out.

Given the .75 multiplier below, we're essentially running all bots now at 1.5X over-capacity relative to physical CPUs.

Longer term, when we look at more sophisticated scheduling, it might be best to have our Intel bots use their _physical_ CPU count as the input to any math we do, so that the math works out across architectures.

geoffreygaren reviewed (approved):
r=me

ryanhaddad committed:
[webkitpy] Increase the default number of child processes spawned for arm64 Macs
https://bugs.webkit.org/show_bug.cgi?id=248625
rdar://102873021

Reviewed by Jonathan Bedard and Geoffrey Garen.

arm64 Mac hardware can handle more WebKitTestRunner processes than the
default number we calculate. Local testing suggests that doubling the count
yields an 18 minute speedup in overall test runtime on M1 hardware, so lets
hardcode this increase until we implement a more sophisticated test scheduling
mechanism.

* Tools/Scripts/webkitpy/port/mac.py:
(MacPort.default_child_processes):

Canonical link: https://commits.webkit.org/257754@main
